File: Programlama Dilllerinin Prensipleri/odev2/KISIM1/RASTGELEKISIURET/src/RastgeleKisiUret/Kisi.java
/**
*
* @author Duhan UZUN, duhan.uzun@ogr.sakarya.ogr.edu.tr - Kadir ÇELİK, kadir.celik6@ogr.sakarya.edu.tr
* @since 20.04.2020
* <p>
* Rastgele kişi üreten ve ürettiği kişilerin kimlik numaralarını ve IMEI numaralarını kontrol eden kütüphanenin kişi kısmı
* </p>
*/

package RastgeleKisiUret;

//kişimizin verilerini bu sınıfta tutuyoruz.
public class Kisi 
{
    //kimlikno ve telefon türünde bir değişken belirtiyoruz
    private KimlikNo kimlikNo;
    private Telefon telefonNo;
    private String isim;
    private String soyisim;
    private int yas;
   

    public String getIsim() {
        return isim;
    }

    public void setIsim(String isim) {
        this.isim = isim;
    }
   
    public KimlikNo getKimlikNo() 
    {
        return kimlikNo;
    }

    public void setKimlikNo(KimlikNo kimlikNo) 
    {
        this.kimlikNo = kimlikNo;
    }

    public String getSoyisim() 
    {
        return soyisim;
    }

    public void setSoyisim(String soyisim) 
    {
        this.soyisim = soyisim;
    }

    public int getYas() {
        return yas;
    }

    public void setYas(int yas) 
    {
        this.yas = yas;
    }

    public Telefon getTelefonNo() 
    {
        return telefonNo;
    }

    public void setTelefonNo(Telefon telefonNo) 
    {
        this.telefonNo = telefonNo;
    }

}

File: Programlama Dilllerinin Prensipleri/odev2/KISIM2/src/Kisi.c
/*
*
* @author Duhan UZUN, duhan.uzun@ogr.sakarya.edu.tr - Kadir ÇELİK, kadir.celik6@ogr.sakarya.edu.tr
* @since 26.04.2020
* <p>
* Rastgele kişi üreten ve ürettiği kişilerin kimlik numaralarını ve IMEI numaralarını kontrol eden programın kişi kısmının kaynak dosyası
* </p>
*/


#include "Kisi.h"

//Kisi yapısını oluşturacak fonksiyon yazılıyor.
Kisi KisiOlustur()
{	
	Kisi this;
	//Kisi için bellekte gerekli yer ayrılıyor
	this = (Kisi)malloc(sizeof(struct KISI));
	
	//Fonksiyonlara ilgili göstericiler atanıyor
	this->yoket = &kisiYoket;	
}

//Kisi yapısını yok eden metod yazılıyor
void kisiYoket(Kisi this)
{	
	if(this == NULL) return;
	this->kimlikNo->kyoket(this->kimlikNo);
	this->telefon->yoket(this->telefon);
	free(this);
}